### PR TITLE
[Swagger Linter Migration] GetInOperationName (origin)

### PR DESCRIPTION
**Pull request by msyyc**
## Original Swagger linter

[`GetInOperationName`](https://github.com/Azure/azure-openapi-validator/blob/6243cb01c16c7535cd3b8df6f45fbeb3c095ed7f/packages/rulesets/src/spectral/functions/get-in-operation-name.ts) verifies ARM GET `operationId` naming.

The original rule performs these checks:

- inspects resolved OpenAPI 2 `operationId` values under GET operations in both `paths` and `x-ms-paths`;
- accepts IDs that begin with `Get` or `List`;
- accepts noun-prefixed IDs whose verb after an underscore begins with `Get` or `List`;
- applies both patterns case-sensitively; and
- reports every other non-empty string at the emitted `operationId` path.

## How the Swagger linter works

The Azure ruleset selects `$[paths,'x-ms-paths'].*[get][?(@property === 'operationId')]` and passes each resolved scalar to a Spectral function. The function returns no result for an empty or non-string value, then applies `^(\w+)_(Get|List)` and `^(Get|List)`. A failure is attached to the selected JSON path, so reused or multiply emitted operations can produce occurrence-level diagnostics.

The migration preserves the case-sensitive naming contract rather than the previous TypeSpec implementation's lowercase source-name approximation. Empty/non-string OpenAPI values do not require a TypeSpec exemption because AutoRest resolves a string operation ID for an emitted semantic operation.

## How the migrated TypeSpec linter works

The TypeSpec rule visits semantic operations, resolves their HTTP shape with `getHttpOperation`, and checks only GET operations. It ignores generic template declarations and instances that have no emitted operation while continuing to check concrete operations created from templates.

For each operation, it reproduces AutoRest operation-ID resolution: explicit `@operationId` wins; otherwise one AutoRest-scoped TCGC context supplies `@clientName` and `@clientLocation` overrides, including AutoRest's falsy empty-location behavior. Interface/namespace prefixes and underscore-segment capitalization match the emitter. The original case-sensitive regexes are then applied to that effective ID, and diagnostics target the authored TypeSpec operation while naming the emitted ID.

## Migration evidence

The detailed [migration evidence](https://github.com/Azure/typespec-azure/blob/feature/lintdiff-get-in-operation-name/packages/typespec-lintdiff/test/fixtures/GetInOperationName/migration.md) records focused violating/compliant fixtures, report reconciliation, code-backed DataMigration/template/TCGC gap examples, full project lists, diagnostic identities, selected-version policy, compile failures, and remaining uncertainty.

The final pinned full corpus run reached 26/26 same-project overlap with no one-sided projects and 39 aligned diagnostics from each engine. Six TypeSpec compile failures remain explicitly excluded from behavioral comparison. Independent review also identified the empty `@clientLocation` edge case; the fix has a dedicated fixture and real-service regression checks.

### CODE REVIEW

File: packages/typespec-lintdiff/test/fixtures/GetInOperationName/client-name-location-compliant/main.tsp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    @statusCode statusCode: 200;
+    @body definitions: string[];
+  };
+}
+
+@@clientLocation(
+  ScopeAccessReviewHistoryDefinitionInstances.listByScope,
+  "ScopeAccessReviewHistoryDefinitions"
+);
+@@clientName(ScopeAccessReviewHistoryDefinitionInstances.listByScope, "List");
```
[the remark this answers is not in the document]
Comment (msyyc) [reply packages/typespec-lintdiff/test/fixtures/GetInOperationName/client-name-location-compliant/main.tsp]: Fixed in `82e0bac7e80147673f35c8a926d725ae70882299`. The authored operation is now `fetchByScope`, while `@clientName(..., "List")` remains, so this fixture specifically exercises client-name resolution.

File: packages/typespec-lintdiff/test/fixtures/GetInOperationName/migration.md [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+| [External coverage snapshot](../../../docs/coverage_old.md)         | The checked-in snapshot links to [the source gist](https://gist.github.com/catalinaperalta/b2e7d29a33b4b451bcfcc87e8314565a), but does not record its generation date, spec commit, or generator revision. It reports 450 compiled projects and 210 validator rules. | `lint`; 26 validator projects fired; 24 local-lint projects credited; 0 official projects; 92.3%. Individual unmatched projects cannot be reconstructed from this aggregate row.                              |
+| [Observed lint-diff coverage](../../../specs/coverage-breakdown.md) | Full run generated 2026-08-26 from specs commit `f6b53f105b95da05276530a0754a1c71b4f16397`. The retained validator dataset contains 468 ARM projects and their selected latest API versions. TypeSpec assessed 462 projects; six failed compilation.                 | `production`; `lint`; 26 validator projects; 26 TypeSpec projects; 26 overlap; 0 validator-only; 0 TypeSpec-only; 39 validator diagnostics; 39 assessed TypeSpec diagnostics; 100% observed project coverage. |
```
[the remark this answers is not in the document]
Comment (msyyc) [reply packages/typespec-lintdiff/test/fixtures/GetInOperationName/migration.md]: Fixed in `82e0bac7e80147673f35c8a926d725ae70882299`. The migration evidence now links to a retained `corpus-evidence.json` artifact containing the dated run metadata, project sets, and 26/26 project plus 39/39 diagnostic totals instead of relying on the stale aggregate reports.
